Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_baking.py

```diff
@@ -0,0 +1,1195 @@
+            if c in bsdf.inputs:
+                return c
+
+        return None
+
+    def bakeChannel(self, context, obj, mat, bsdf, matOutput, channel):
+        canBakeBSDF = self.canBsdfBeBaked(bsdf)
+
+        mat = self.prepareBake(context, obj, bsdf, mat, canBakeBSDF)
+
+        # Ideally, we should bake the BSDF inputs:
+        if canBakeBSDF:
+            bakedImageNode = self.bakeChannelInputsDirectly(context, obj, mat, bsdf, matOutput, channel)
+        else:
+            bakedImageNode = self.bakeWithFallback(context, obj, mat, channel)
+
+        if not bakedImageNode:
+            return mat
+
+        # Connect baked image to baked bsdf node:
+        node_tree = mat.node_tree
+        bsdf = node_tree.nodes[mat[BAKED_MATERIAL_SHADER]]
+        inputChannelName = self.getBSDFChannelInputName(bsdf, channel)
+
+        # Connect baked image to the input now:
+        if inputChannelName == 'Normal':
+            normalMapNode = node_tree.nodes.new("ShaderNodeNormalMap")
+            normalMapNode.location = (bakedImageNode.location[0], bakedImageNode.location[1])
+            bakedImageNode.location = (bakedImageNode.location[0] - 300, bakedImageNode.location[1])
+
+            node_tree.links.new(bakedImageNode.outputs[0], normalMapNode.inputs['Color'])
+            node_tree.links.new(normalMapNode.outputs[0], bsdf.inputs[inputChannelName])
+        else:
+            node_tree.links.new(bakedImageNode.outputs[0], bsdf.inputs[inputChannelName])
+
+        return mat
+
+    def execute(self, context):
+        if not os.access(context.scene.meshsync_bake_settings.bakedTexturesPath, os.W_OK):
+            self.report({'WARNING'}, "The folder to save baked textures to does not exist!")
+            self.stop(context)
+            return {'CANCELLED'}
+
+        self.startTime = time.time()
+        self.context = context
+        self.bakeTask = self.bake()
+
+        return {'RUNNING_MODAL'}
+
+    def stop(self, context):
+        wm = context.window_manager
+        wm.event_timer_remove(self.timer)
+
+    def invoke(self, context, event):
+        wm = context.window_manager
+        self.timer = wm.event_timer_add(0, window=context.window)
+        context.window_manager.modal_handler_add(self)
+
+        # Store area for later use, these won't exist on modal timer callbacks:
+        self.area = context.area
+
+        return self.execute(context)
+
+    def modal(self, context, event):
+        # Allow cancellation by pressing escape:
+        if event.type == 'ESC':
+            self.stop(context)
+            self.incrementProgress(context, "Baking cancelled by user", reset=True)
+            return {'FINISHED'}
+
+        # Refresh context each run:
+        self.context = context
+        for _ in self.bakeTask:
+            return {'RUNNING_MODAL'}
+
+        msb_log(f"Finished baking. Time taken: {datetime.timedelta(seconds=(time.time() - self.startTime))}",
+                LogLevel.ERROR)
```

Comment:
Is this supposed to be an error?

Reply:
It's the highest log level, to ensure the message always shows.

Reply:
Does it say "ERROR" in the message though? Or does it just show up red or something?

Reply:
The log level just defines whether the logs should show or not. It's set at the top of the file. For releases it should be set to ERROR, for debugging VERBOSE.